### ISSUE TITLE
Gurobi integration

### ISSUE THREAD

**Issue by janmotl** [optimatika/ojAlgo-extensions#2]
With the following code: 
```
       Variable[] objective = new Variable[]{
                new Variable("X1").weight(0.8),
                new Variable("X2").weight(0.2),
                new Variable("X3").weight(0.7),
                new Variable("X4").weight(0.3),
                new Variable("X5").weight(0.6),
                new Variable("X6").weight(0.4)};
        
        ExpressionsBasedModel.addIntegration(SolverGurobi.INTEGRATION);
        ExpressionsBasedModel model = new ExpressionsBasedModel(objective);

        model.addExpression("C1").set(0, 1).set(2, 1).set(4, 1).level(23);
        model.addExpression("C2").set(1, 1).set(3, 1).set(5, 1).level(23);
        model.addExpression("C3").set(0, 1).lower(10);
        model.addExpression("C4").set(2, 1).lower(8);
        model.addExpression("C5").set(4, 1).lower(5);

        Result result = model.maximise();

        // A valid solution of 25.8 can be produced with:
        //     X1=10, X2=0, X3=8, X4=0, X5=5, X6=23
        Assert.assertEquals(25.8, result.getValue(), 0.001);
```
I am getting: `gurobi.GRBException: Unable to retrieve attribute 'ObjVal'`. 

The code works without Gurobi integration. And Gurobi 7.5 by itself works as well. Tested with Java 8 on OS X.

**Comment by apete** [comment]
I never actually tested the Gurobi integration. It was written at the same time as the MOSEK and CPLEX integrations, but was never actually used.

If you want to try and fix it I'll help, but I don't have Gurobi license.

It's probably a good idea to look at how the MOSEK and CPLEX integrations are done.

**Comment by apete** [comment]
I made some changes to SolverGurobi - made it more similar to SolverCPLEX and fixed some problems. (Probably didn't fix your issue.)

I asked Gurobi for a temporary license so I could finish this, but I haven't heard back from them. With a license I have no way to test. Someone else (someone with a license) needs to do that.

**Comment by janmotl** [comment]
I tried the updated class with 44.0.1-SNAPSHOT of ojAlgo. The error is the same, but maybe a complete stack trace could help:
```
Optimize a model with 2 rows, 6 columns and 6 nonzeros
Coefficient statistics:
  Matrix range     [1e+00, 1e+00]
  Objective range  [2e-01, 8e-01]
  Bounds range     [5e+00, 1e+01]
  RHS range        [2e+01, 2e+01]
Presolve removed 1 rows and 1 columns
Presolve time: 0.00s

Solved in 0 iterations and 0.00 seconds
Infeasible or unbounded model
gurobi.GRBException: Unable to retrieve attribute 'ObjVal'
	at gurobi.GRBModel.get(GRBModel.java:3551)
	at org.ojalgo.optimisation.external.SolverGurobi.solve(SolverGurobi.java:242)
	at org.ojalgo.optimisation.ExpressionsBasedModel.solve(ExpressionsBasedModel.java:764)
	at org.ojalgo.optimisation.ExpressionsBasedModel.maximise(ExpressionsBasedModel.java:626)
	at SolverTest.testMath286(SolverTest.java:35)

java.lang.NullPointerException
	at java.util.Objects.requireNonNull(Objects.java:203)
	at org.ojalgo.ProgrammingError.throwIfNull(ProgrammingError.java:94)
	at org.ojalgo.optimisation.Optimisation$Result.<init>(Optimisation.java:300)
	at org.ojalgo.optimisation.external.SolverGurobi.solve(SolverGurobi.java:255)
	at org.ojalgo.optimisation.ExpressionsBasedModel.solve(ExpressionsBasedModel.java:764)
	at org.ojalgo.optimisation.ExpressionsBasedModel.maximise(ExpressionsBasedModel.java:626)
	at SolverTest.testMath286(SolverTest.java:35)
```
But I do not fell to be competent to debug it.

If you are from an academia, then academic license of Gurobi is without waiting - you just have to activate the license from an academic network.

**Comment by apete** [comment]
Think maybe I just forgot to update the GRBModel after adding the constraints. Try again.

(The academic license doesn't apply to me.)

**Comment by janmotl** [comment]
Once I add constraints:
```
        model.addExpression("C6").set(1, 1).lower(0);   // X2 >= 0
        model.addExpression("C7").set(3, 1).lower(0);   // X4 >= 0
        model.addExpression("C8").set(5, 1).lower(0);   // X6 >= 0
```
It works. The problem was that during the definition of the variables the lower bounds for {X2, X4, X6} were set to `Double.NEGATIVE_INFINITY`. And once you permit negative values, the problem is `Infeasible or unbounded model` and `ObjVal` is not set.

It looks like both, ojAlgo and Gurobi, assume that variables are, if not told otherwise, non-negative. But why is then negative infinity passed as the lower bound?

**Comment by apete** [comment]
Good to hear that something works, but ojAlgo makes no assumptions on restrictions (non-negative) of variables (and I'm pretty sure neither does Gurobi). That part sounds confusing.

Could you do a few more tests (be sure to use the latest source code) with some variations to check if you get the same results with or without the Gurobi integration.

**Comment by janmotl** [comment]
http://www.gurobi.com/documentation/7.0/refman/variables.html 
> In mathematical programming, the convention is that variables are non-negative unless stated otherwise, so if you don't explicitly provide bounds for a variable, you should assume that the lower bound is 0 and the upper bound is infinite. 

I will check the changes.

**Comment by janmotl** [comment]
ojAlgo seems to make the same assumption. With the original formulation of the problem, ojAlgo returns 25.8. But if I add constraint:
```
model.addExpression("C6").set(1, 1).upper(-10);
```
ojAlgo returns 27.8.

**Comment by apete** [comment]
ojAlgo does not make that assumption, I know that ;-) With the original problem formulation ojAlgo outputs this Optimisation.Result:
```
UNBOUNDED 25.8 @ [10.0, 0.0, 8.0, 0.0, 5.0, 23.0]
```
The fact that a problem/model is unbounded doesn't mean it's impossible to find a feasible solution, and that's what ojAlgo does.

**Comment by janmotl** [comment]
I just noticed that as well :)

**Comment by janmotl** [comment]
I executed some unit tests with Gurobi solver. LinearProblems finished without any issue. But there was a few for IntegerProblems:

In testP20130225:
``` 
Optimize a model with 57 rows, 569 columns and 1539 nonzeros
Coefficient statistics:
  Matrix range     [1e-01, 1e+01]
  Objective range  [1e-01, 5e+01]
  Bounds range     [1e+00, 1e+00]
  RHS range        [5e-01, 9e+00]
Presolve removed 0 rows and 28 columns
Presolve time: 0.00s
Presolved: 57 rows, 541 columns, 1511 nonzeros

Iteration    Objective       Primal Inf.    Dual Inf.      Time
       0    2.3350000e+02   1.132500e+01   0.000000e+00      0s
      28    2.5560000e+02   0.000000e+00   0.000000e+00      0s

Solved in 28 iterations and 0.01 seconds
Optimal objective  2.556000000e+02

junit.framework.AssertionFailedError: Solution To Relaxed Problem Not Valid!
```

In testP20140819:
``` 
Optimize a model with 8 rows, 16 columns and 56 nonzeros
Variable types: 0 continuous, 16 integer (0 binary)
Coefficient statistics:
  Matrix range     [4e-01, 7e+00]
  Objective range  [3e+00, 3e+00]
  Bounds range     [4e+02, 4e+02]
  RHS range        [4e+00, 2e+04]
Presolve time: 0.00s

Explored 0 nodes (0 simplex iterations) in 0.00 seconds
Thread count was 1 (of 4 available processors)

Solution count 0

Model is infeasible
Best objective -, best bound -, gap -

junit.framework.AssertionFailedError: INFEASIBLE 0.0 @ [0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0]
```

In testP20160701:
``` 
Optimize a model with 32 rows, 35 columns and 120 nonzeros
Variable types: 5 continuous, 30 integer (30 binary)
Coefficient statistics:
  Matrix range     [1e+00, 6e+00]
  Objective range  [1e+00, 4e+00]
  Bounds range     [1e+00, 1e+00]
  RHS range        [1e+00, 5e+00]
Presolve removed 0 rows and 1 columns
Presolve time: 0.01s

Explored 0 nodes (0 simplex iterations) in 0.01 seconds
Thread count was 1 (of 4 available processors)

Solution count 0

Model is infeasible
Best objective -, best bound -, gap -

junit.framework.AssertionFailedError: INFEASIBLE 0.0 @ [0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0]
```
In testP20150127full:
``` 
java.lang.NullPointerException
	at gurobi.GRBModel.setObjective(GRBModel.java:892)
	at org.ojalgo.optimisation.external.SolverGurobi$1.build(SolverGurobi.java:126)
	at org.ojalgo.optimisation.external.SolverGurobi$1.build(SolverGurobi.java:83)
	at org.ojalgo.optimisation.ExpressionsBasedModel.solve(ExpressionsBasedModel.java:762)
	at org.ojalgo.optimisation.ExpressionsBasedModel.minimise(ExpressionsBasedModel.java:635)
```

In testP20130409a:
``` 
Optimize a model with 3 rows, 3 columns and 8 nonzeros
Variable types: 1 continuous, 2 integer (0 binary)
Coefficient statistics:
  Matrix range     [1e-02, 5e+01]
  Objective range  [1e+00, 1e+00]
  Bounds range     [4e+03, 4e+03]
  RHS range        [2e+03, 2e+03]
Presolve removed 1 rows and 2 columns
Presolve time: 0.00s

Explored 0 nodes (0 simplex iterations) in 0.00 seconds
Thread count was 1 (of 4 available processors)

Solution count 0

Model is infeasible
Best objective -, best bound -, gap -

junit.framework.AssertionFailedError: Solution Not Correct, large norm differences 4994.066979927344 !<< 4994.066979927344
```

In testP20130409b:
``` 
Optimize a model with 5 rows, 4 columns and 17 nonzeros
Variable types: 1 continuous, 3 integer (0 binary)
Coefficient statistics:
  Matrix range     [1e-02, 5e+01]
  Objective range  [1e+00, 1e+00]
  Bounds range     [0e+00, 0e+00]
  RHS range        [2e+03, 2e+03]
Presolve removed 3 rows and 2 columns
Presolve time: 0.00s

Explored 0 nodes (0 simplex iterations) in 0.00 seconds
Thread count was 1 (of 4 available processors)

Solution count 0

Model is infeasible
Best objective -, best bound -, gap -

junit.framework.AssertionFailedError: Solution Not Correct, large norm differences 5328.333229067044 !<< 5328.333229067044
```

There are also some issues in ConvexProblems:

In testP20080117:
``` 
Optimize a model with 1 rows, 9 columns and 9 nonzeros
Model has 45 quadratic objective terms
Coefficient statistics:
  Matrix range     [1e+00, 1e+00]
  Objective range  [3e+00, 7e+01]
  QObjective range [1e-01, 1e+02]
  Bounds range     [2e-02, 8e-01]
  RHS range        [1e+00, 1e+00]
gurobi.GRBException: Objective Q not PSD (diagonal adjustment of 4.7e+00 would be required)
	at gurobi.GRBModel.optimize(GRBModel.java:1027)
	at org.ojalgo.optimisation.external.SolverGurobi.solve(SolverGurobi.java:241)
	at org.ojalgo.optimisation.ExpressionsBasedModel.solve(ExpressionsBasedModel.java:764)
	at org.ojalgo.optimisation.ExpressionsBasedModel.minimise(ExpressionsBasedModel.java:635)
	at org.ojalgo.optimisation.convex.ConvexProblems.doEarly2008(ConvexProblems.java:181)
```

In testP20111129:
``` 
Optimize a model with 4 rows, 4 columns and 12 nonzeros
Model has 3 quadratic objective terms
Coefficient statistics:
  Matrix range     [1e+00, 4e+00]
  Objective range  [1e+00, 4e+00]
  QObjective range [1e+00, 4e+00]
  Bounds range     [1e+00, 1e+02]
  RHS range        [1e+00, 6e+00]
Presolve removed 2 rows and 2 columns
Presolve time: 0.00s
Presolved: 2 rows, 2 columns, 4 nonzeros
Presolved model has 3 quadratic objective terms
Ordering time: 0.00s

Barrier statistics:
 Free vars  : 1
 AA' NZ     : 3.000e+00
 Factor NZ  : 6.000e+00
 Factor Ops : 1.400e+01 (less than 1 second per iteration)
 Threads    : 1

                  Objective                Residual
Iter       Primal          Dual         Primal    Dual     Compl     Time
   0  -2.79988454e+04 -2.56995000e+04  9.76e+02 3.48e-06  1.01e+06     0s
   1  -3.21723183e+03 -3.23125285e+04  1.98e+02 4.49e-07  1.36e+05     0s
   2   4.12241209e+03 -3.00623670e+04  9.68e-05 1.93e-12  6.84e+03     0s
   3   2.05113937e+03 -2.12814946e+03  6.32e-06 1.14e-13  8.36e+02     0s
   4   3.26397825e+02 -5.16259136e+02  6.32e-12 2.84e-14  1.69e+02     0s
   5   4.64409342e+01 -5.60640071e+01  2.89e-15 5.68e-14  2.05e+01     0s
   6   3.79971471e+00 -9.30747831e+00  2.00e-15 1.07e-14  2.62e+00     0s
   7   1.11327222e-02 -3.46272752e-01  1.33e-15 1.07e-14  7.15e-02     0s
   8   1.17249278e-05 -3.46644201e-04  3.77e-15 9.77e-15  7.17e-05     0s
   9   1.17365957e-08 -3.46989925e-07  2.66e-15 1.42e-14  7.17e-08     0s
  10   1.22284405e-11 -3.47309737e-10  2.66e-15 1.78e-14  7.18e-11     0s

Barrier solved model in 10 iterations and 0.00 seconds
Optimal objective 1.22284405e-11


junit.framework.AssertionFailedError: double != double 
Expected :-5.281249989
Actual   :0.0
```

In testP20150720:
``` 
Optimize a model with 6 rows, 8 columns and 16 nonzeros
Model has 4 quadratic objective terms
Coefficient statistics:
  Matrix range     [3e-03, 1e+02]
  Objective range  [1e+00, 1e+00]
  QObjective range [2e+00, 2e+00]
  Bounds range     [4e-01, 4e+04]
  RHS range        [6e+01, 5e+04]
Presolve removed 5 rows and 8 columns
Presolve time: 0.00s

Barrier solved model in 0 iterations and 0.00 seconds
Model is infeasible
Optimize a model with 1 rows, 4 columns and 4 nonzeros
Model has 4 quadratic objective terms
Coefficient statistics:
  Matrix range     [1e+00, 1e+00]
  Objective range  [6e+03, 5e+04]
  QObjective range [2e+00, 2e+00]
  Bounds range     [5e+03, 4e+04]
  RHS range        [5e+04, 5e+04]
Presolve time: 0.00s
Presolved: 1 rows, 4 columns, 4 nonzeros
Presolved model has 4 quadratic objective terms
Ordering time: 0.00s

Barrier statistics:
 AA' NZ     : 0.000e+00
 Factor NZ  : 1.000e+00
 Factor Ops : 1.000e+00 (less than 1 second per iteration)
 Threads    : 1

                  Objective                Residual
Iter       Primal          Dual         Primal    Dual     Compl     Time
   0   4.15826932e+08  2.38234130e+09  1.81e+04 1.03e+04  5.63e+07     0s
   1   8.15132240e+08  8.73852070e+08  0.00e+00 1.03e-02  6.52e+06     0s
   2   8.16748617e+08  8.18608529e+08  0.00e+00 8.58e-05  2.07e+05     0s
   3   8.17121046e+08  8.17160757e+08  0.00e+00 8.55e-11  4.41e+03     0s
   4   8.17148822e+08  8.17148908e+08  0.00e+00 0.00e+00  9.56e+00     0s
   5   8.17148897e+08  8.17148897e+08  0.00e+00 0.00e+00  9.57e-03     0s
   6   8.17148897e+08  8.17148897e+08  0.00e+00 0.00e+00  9.58e-06     0s
   7   8.17148897e+08  8.17148897e+08  0.00e+00 0.00e+00  9.59e-09     0s

Barrier solved model in 7 iterations and 0.00 seconds
Optimal objective 8.17148897e+08

Optimize a model with 16 rows, 45 columns and 88 nonzeros
Model has 82 quadratic objective terms
Coefficient statistics:
  Matrix range     [1e+00, 1e+00]
  Objective range  [1e+11, 1e+11]
  QObjective range [4e-01, 2e+02]
  Bounds range     [3e+02, 6e+04]
  RHS range        [3e+02, 9e+05]
Warning: Model contains large objective coefficients
         Consider reformulating model or setting NumericFocus parameter
         to avoid numerical issues.
Presolve time: 0.00s

Barrier solved model in 0 iterations and 0.00 seconds
Model is infeasible or unbounded
Optimize a model with 6 rows, 8 columns and 16 nonzeros
Model has 4 quadratic objective terms
Coefficient statistics:
  Matrix range     [3e-03, 1e+02]
  Objective range  [1e+00, 1e+00]
  QObjective range [2e+00, 2e+00]
  Bounds range     [4e-01, 4e+04]
  RHS range        [6e+01, 5e+04]
Presolve removed 5 rows and 8 columns
Presolve time: 0.00s

Barrier solved model in 0 iterations and 0.00 seconds
Model is infeasible

junit.framework.AssertionFailedError: INFEASIBLE -0.3302946228035407 @ [0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, -0.5747126436781609]

```

In testInfeasibleCase:
``` 
Optimize a model with 1 rows, 3 columns and 3 nonzeros
Model has 3 quadratic objective terms
Coefficient statistics:
  Matrix range     [1e+00, 1e+00]
  Objective range  [1e+00, 3e+00]
  QObjective range [2e+00, 2e+01]
  Bounds range     [1e+00, 2e+00]
  RHS range        [2e+00, 2e+00]
Presolve removed 1 rows and 3 columns
Presolve time: 0.00s
Presolve: All rows and columns removed

Barrier solved model in 0 iterations and 0.00 seconds
Optimal objective 7.10050288e+01

junit.framework.AssertionFailedError
	at junit.framework.Assert.fail(Assert.java:55)
	at junit.framework.Assert.assertTrue(Assert.java:22)
	at junit.framework.Assert.assertFalse(Assert.java:39)
	at junit.framework.Assert.assertFalse(Assert.java:47)
	at org.ojalgo.TestUtils.assertFalse(TestUtils.java:336)
```

I did not try other unit tests.

**Comment by apete** [comment]
That's a lot of problems. What did you do? Did you run all ojAlgo unit test on Gurobi instead?

Some of these problems may just be rounding errors. Inspecting the solution you may find it acceptable.

Since I don't have a Gurobi license and can't run any of these tests, I can only help with small models where you have essentially found the problem. (I can help with the fix.)

**Comment by janmotl** [comment]
> What did you do? Did you run all ojAlgo unit test on Gurobi instead?

Yes, but I modified just 3 classes. 

> Some of these problems may just be rounding errors. Inspecting the solution you may find it acceptable.

I agree.

> Since I don't have a Gurobi license and can't run any of these tests, I can only help with small models where you have essentially found the problem. (I can help with the fix.)

I understand. Since my resources are limited, I propose to close the issue as the original problem was solved. If I encounter some issue where I know where is the problem, I will create a new issue.

Thank you for your support.

**Comment by apete** [comment]
Sounds good to me.

**Comment by apete** [comment]
Just fixed a bug in the CPLEX integration, and the same problem (same code mistake) was also in the Gurobi integration. Maybe all those other tests work better now.

**Comment by janmotl** [comment]
No improvement was observed. But now one more unit test, testP20081119 in ConvexProblems, is failing:
```
Optimize a model with 4 rows, 4 columns and 12 nonzeros
Model has 3 quadratic objective terms
Coefficient statistics:
  Matrix range     [1e+00, 4e+00]
  Objective range  [1e+00, 4e+00]
  QObjective range [1e+00, 4e+00]
  Bounds range     [1e+00, 1e+02]
  RHS range        [1e+00, 6e+00]
Presolve removed 2 rows and 2 columns
Presolve time: 0.01s
Presolved: 2 rows, 2 columns, 4 nonzeros
Presolved model has 3 quadratic objective terms
Ordering time: 0.00s

Barrier statistics:
 Free vars  : 1
 AA' NZ     : 3.000e+00
 Factor NZ  : 6.000e+00
 Factor Ops : 1.400e+01 (less than 1 second per iteration)
 Threads    : 1

                  Objective                Residual
Iter       Primal          Dual         Primal    Dual     Compl     Time
   0  -2.79988454e+04 -2.56995000e+04  9.76e+02 3.48e-06  1.01e+06     0s
   1  -3.21723183e+03 -3.23125285e+04  1.98e+02 4.49e-07  1.36e+05     0s
   2   4.12241209e+03 -3.00623670e+04  9.68e-05 1.93e-12  6.84e+03     0s
   3   2.05113937e+03 -2.12814946e+03  6.32e-06 1.14e-13  8.36e+02     0s
   4   3.26397825e+02 -5.16259136e+02  6.32e-12 2.84e-14  1.69e+02     0s
   5   4.64409342e+01 -5.60640071e+01  2.89e-15 5.68e-14  2.05e+01     0s
   6   3.79971471e+00 -9.30747831e+00  2.00e-15 1.07e-14  2.62e+00     0s
   7   1.11327222e-02 -3.46272752e-01  1.33e-15 1.07e-14  7.15e-02     0s
   8   1.17249278e-05 -3.46644201e-04  3.77e-15 9.77e-15  7.17e-05     0s
   9   1.17365957e-08 -3.46989925e-07  2.66e-15 1.42e-14  7.17e-08     0s
  10   1.22284405e-11 -3.47309737e-10  2.66e-15 1.78e-14  7.18e-11     0s

Barrier solved model in 10 iterations and 0.01 seconds
Optimal objective 1.22284405e-11


junit.framework.AssertionFailedError: double != double 
Expected :-5.281249989
Actual   :0.0
```
The original message used to be:
```
Optimize a model with 4 rows, 4 columns and 12 nonzeros
Model has 3 quadratic objective terms
Coefficient statistics:
  Matrix range     [1e+00, 4e+00]
  Objective range  [1e+00, 4e+00]
  QObjective range [1e+00, 4e+00]
  Bounds range     [1e+00, 1e+02]
  RHS range        [1e+00, 6e+00]
Presolve removed 2 rows and 2 columns
Presolve time: 0.01s
Presolved: 2 rows, 2 columns, 4 nonzeros
Presolved model has 3 quadratic objective terms
Ordering time: 0.00s

Barrier statistics:
 Free vars  : 1
 AA' NZ     : 3.000e+00
 Factor NZ  : 6.000e+00
 Factor Ops : 1.400e+01 (less than 1 second per iteration)
 Threads    : 1

                  Objective                Residual
Iter       Primal          Dual         Primal    Dual     Compl     Time
   0  -2.79988454e+04 -2.56995000e+04  9.76e+02 3.48e-06  1.01e+06     0s
   1  -3.21723183e+03 -3.23125285e+04  1.98e+02 4.49e-07  1.36e+05     0s
   2   4.12241209e+03 -3.00623670e+04  9.68e-05 1.93e-12  6.84e+03     0s
   3   2.05113937e+03 -2.12814946e+03  6.32e-06 1.14e-13  8.36e+02     0s
   4   3.26397825e+02 -5.16259136e+02  6.32e-12 2.84e-14  1.69e+02     0s
   5   4.64409342e+01 -5.60640071e+01  2.89e-15 5.68e-14  2.05e+01     0s
   6   3.79971471e+00 -9.30747831e+00  2.00e-15 1.07e-14  2.62e+00     0s
   7   1.11327222e-02 -3.46272752e-01  1.33e-15 1.07e-14  7.15e-02     0s
   8   1.17249278e-05 -3.46644201e-04  3.77e-15 9.77e-15  7.17e-05     0s
   9   1.17365957e-08 -3.46989925e-07  2.66e-15 1.42e-14  7.17e-08     0s
  10   1.22284405e-11 -3.47309737e-10  2.66e-15 1.78e-14  7.18e-11     0s

Barrier solved model in 10 iterations and 0.01 seconds
Optimal objective 1.22284405e-11


junit.framework.AssertionFailedError: double != double 
Expected :-5.281249989
Actual   :0.0
```
